plugins/packages/restapi/lib/index.ts

```diff
@@ -155,6 +156,45 @@ export default class RestapiQueryService implements QueryService {
     };
   }
 
+  fetchHttpsCertsForCustomCA(sourceOptions: any) {
+    let httpsParams: any = {};
+    switch (sourceOptions.ssl_certificate) {
+      case 'ca_certificate':
+        httpsParams = {
+          https: {
+            certificateAuthority: [sourceOptions.ca_cert],
+          },
+        };
+        break;
+      case 'client_certificate':
+        httpsParams = {
+          https: {
+            certificateAuthority: [sourceOptions.ca_cert],
+            key: [sourceOptions.client_key],
+            certificate: [sourceOptions.client_cert],
+          },
+        };
+        break;
+      default:
+        break;
+    }
+
+    if (process.env.NODE_EXTRA_CA_CERTS) {
```

Comment:
could be refactored as:

```
if (process.env.NODE_EXTRA_CA_CERTS) {
  const certificateAuthority = [
    ...tls.rootCertificates,
    readFileSync(process.env.NODE_EXTRA_CA_CERTS),
  ].join('\n');

  httpsParams.https = httpsParams.https || {};
  httpsParams.https.certificateAuthority = certificateAuthority;
}
```

Reply:
REST-API plugin, When **SSL** is enabled, user will be providing **ca_cert**, along with that when NODE_EXTRA_CA_CERTS was given in env file, then in that case we need to concat NODE_EXTRA_CA_CERTS along with **ca_cert** provided by the user. So one more condition is necessary. We can refactor like
`const certificateAuthority = [
        ...tls.rootCertificates,
        readFileSync(process.env.NODE_EXTRA_CA_CERTS)
      ]`
      
      `httpsParams.https = httpsParams.https || {}
      if (httpsParams.https.certificateAuthority) {
        httpsParams.https.certificateAuthority = httpsParams.https.certificateAuthority.concat(certificateAuthority)
      } else {
        httpsParams.https.certificateAuthority = certificateAuthority;
      }`
      
      But anyhow one more condition is need there

Reply:
got it